src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hidden === false) {
+      victim.health -= 50;
+    }
+
+    if (victim.health === 0) {
+      const onlyAlive = Animal.alive.filter(animal => {
+        return animal.name !== victim.name;
+      });
+
+      Animal.alive = onlyAlive;
```

Comment:
this onlyAlive is redundant
```suggestion
      Animal.alive = Animal.alive.filter(animal => {
        return animal.name !== victim.name;
      });
```